presidio-analyzer/presidio_analyzer/predefined_recognizers/in_gstin_recognizer.py

```diff
@@ -0,0 +1,161 @@
+from typing import Optional, List, Tuple
+from presidio_analyzer import Pattern, PatternRecognizer
+from presidio_analyzer.analyzer_utils import PresidioAnalyzerUtils
+
+
+# from presidio_analyzer.analyzer_utils import PresidioAnalyzerUtils as Utils
+
+
+class InGstinRecognizer(PatternRecognizer):
+    """
+    Recognizes Indian Goods and Services Tax Identification Number ("GSTIN").
+
+    The GSTIN  is a fifteen character alpha-numeric code
+    with the last digit being a check digit calculated using a
+    modified modulus 36 LUHN calculation.
+    This recognizer identifies GSTIN using regex, context words and calculated value.
+    Reference: https://en.wikipedia.org/wiki/Goods_and_Services_Tax_(India),
+               http://idtc-icai.s3.amazonaws.com/download/knowledgeShare18-19/Structure-of-GSTIN.pdf
+
+    :param patterns: List of patterns to be used by this recognizer
+    :param context: List of context words to increase confidence in detection
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    :param replacement_pairs: List of tuples with potential replacement values
+    for different strings to be used during pattern matching.
+    This can allow a greater variety in input, for example by removing dashes or spaces.
+    """
+
+    CONTEXT = [
+        "GSTIN",
+        "GST",
+    ]
+
+    _in_state_tin_codes = {
+        "01": "Jammu and Kashmir",
+        "02": "Himachal Pradesh",
+        "03": "Punjab",
+        "04": "Chandigarh",
+        "05": "Uttarakhand",
+        "06": "Haryana",
+        "07": "Delhi",
+        "08": "Rajasthan",
+        "09": "Uttar Pradesh",
+        "10": "Bihar",
+        "11": "Sikkim",
+        "12": "Arunachal Pradesh",
+        "13": "Nagaland",
+        "14": "Manipur",
+        "15": "Mizoram",
+        "16": "Tripura",
+        "17": "Meghalaya",
+        "18": "Assam",
+        "19": "West Bengal",
+        "20": "Jharkhand",
+        "21": "Orissa",
+        "22": "Chattisgarh",
+        "23": "Madhya Pradesh",
+        "24": "Gujarat",
+        "25": "Daman and Diu",
+        "26": "Dadar and Nagar Haveli",
+        "27": "Maharashtra",
+        "28": "Andhra Pradesh",
+        "29": "Karnataka",
+        "30": "Goa",
+        "31": "Lakshadweep",
+        "32": "Kerala",
+        "33": "Tamil Nadu",
+        "34": "Puducherry",
+        "35": "Anadaman and Nicobar Islands",
+        "36": "Telangana",
+        "37": "Andhra Pradesh (New)",
+    }
+
+    def __init__(
+        self,
+        patterns: Optional[List[Pattern]] = None,
+        context: Optional[List[str]] = None,
+        supported_language: str = "en",
+        supported_entity: str = "IN_GSTIN",
+        replacement_pairs: Optional[List[Tuple[str, str]]] = None,
+        analyzer_utils=PresidioAnalyzerUtils(),
```

Comment:
```suggestion
        analyzer_utils: Optional[PresidioAnalyzerUtils] = None,
```
Mutable objects should not be a default value (https://betterprogramming.pub/the-problem-with-mutable-types-as-default-values-in-functions-in-python-81ca88ff4d91)

Also, since all recognizers now support an `analyzer_utils` parameter, why not pass it during the recognizers instantiation, and only create it if it's missing for this specific recognizer?

This is one line out of two in `RecognizerRegistry` which instantiates the recognizers: 
https://github.com/devopam/presidio/blob/7d6ee388b40baca87809c8d8101b5fed3516cee0/presidio-analyzer/presidio_analyzer/recognizer_registry.py#L140